src/algorithms/string/knuth-morris-pratt/__test__/knuthMorrisPratt.test.js

```diff
@@ -2,6 +2,9 @@ import knuthMorrisPratt from '../knuthMorrisPratt';
 
 describe('knuthMorrisPratt', () => {
   it('should find word position in given text', () => {
+    expect(knuthMorrisPratt('', '')).toBe(0);
+    expect(knuthMorrisPratt('a', '')).toBe(0);
```

Comment:
This test is actually not correct. It says that when I search for an empty word '' in a string 'a' I should find one and this empty word will be accessible at zero index. But `string[0]` will return us `a` which expected but is not correct according to this test. So I assume that we should expect `-1` here instead of `0`.

Reply:
@trekhleb Actually I was understanding it a bit differently.

In my understanding I was expecting `knuthMorrisPratt` to return the same as `String.prototype.indexOf` - *more a less in a sense*.
In other word I considered the output as being: *index within `text` of the first occurrence of `word`* [[official reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf)].

```js
// definition in terms of code
const startIdx = knuthMorrisPratt(text, word);
text.substr(startIdx, word.length) === word // should be true for startIdx !== -1
```

Actually both `"".indexOf("") === 0` and `"a".indexOf("") === 0`.